File: mytool/template_project/timeseries.r
#replace "value" with the actual column name
# Load dataset
data <- read.csv("data.csv")

# Convert to time series
ts_data <- ts(data$value, frequency = 12)

# 1. Trend (moving average)
trend <- filter(ts_data, rep(1/5, 5), sides = 2)

# 2. Exponential smoothing
smooth <- HoltWinters(ts_data, beta=FALSE, gamma=FALSE)

# 3. Forecast
forecast <- predict(smooth, n.ahead = 5)

# Plot
plot(ts_data)
lines(trend, col="red")
lines(smooth$fitted[,1], col="blue")

File: mytool/template_project/timeseries.py
#replace value with actual column names

import pandas as pd
import matplotlib.pyplot as plt

# Load dataset
df = pd.read_csv("data.csv")

# Convert date column
df["date"] = pd.to_datetime(df["date"])
df.set_index("date", inplace=True)

# 1. Trend (rolling mean)
df["trend"] = df["value"].rolling(window=5).mean()

# 2. Exponential smoothing
df["smooth"] = df["value"].ewm(span=5).mean()

# 3. Simple forecasting (shift)
df["forecast"] = df["value"].shift(1)

# Plot
plt.plot(df["value"], label="Original")
plt.plot(df["trend"], label="Trend")
plt.plot(df["smooth"], label="Smoothed")
plt.legend()
plt.show()